frame/contracts/src/lib.rs

```diff
@@ -552,7 +556,8 @@ decl_module! {
 			let result = Self::execute_wasm(origin, &mut gas_meter, |ctx, gas_meter| {
 				ctx.call(dest, value, gas_meter, data)
 			});
-			gas_meter.into_dispatch_result(result, T::WeightInfo::call())
+			let code_len = result.as_ref().map(|r| r.1).unwrap_or(T::MaxCodeSize::get());
```

Comment:
isn't a bit unfair to consider in case the call return an error ?
maybe we can give the code_len also in the error side ?
If the price for MaxCodeSize isn't too high then I guess it doesn't matter to be precise.

same question for other calls.

Reply:
Yes this is a sub optimal solution. This problem also exists for all other entry points. I avoided passing it on error because that would mean to return something like `(Result<T, DispatchError>, u32)` which removed the ability to use `?`. Error handling becomes very annoying.

Reply:
maybe using `Result<(T, u32), (DispatchError, u32)>` and do some map_err everywhere

Reply:
It is really annoying but I guess it has to be done.

Reply:
Done.